app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantingAdapter.kt

```diff
@@ -16,48 +16,97 @@
 
+import android.app.AlertDialog
 import android.content.Context
+import android.util.Log
 import android.view.LayoutInflater
+import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.findNavController
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
+import com.google.samples.apps.sunflower.GardenFragmentDirections
 import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.data.GardenPlantingDao
+import com.google.samples.apps.sunflower.data.GardenPlantingRepository
+import com.google.samples.apps.sunflower.data.Plant
 import com.google.samples.apps.sunflower.data.PlantAndGardenPlantings
 import com.google.samples.apps.sunflower.databinding.ListItemGardenPlantingBinding
+import com.google.samples.apps.sunflower.utilities.InjectorUtils
 import com.google.samples.apps.sunflower.viewmodels.PlantAndGardenPlantingsViewModel
+import com.google.samples.apps.sunflower.viewmodels.PlantDetailViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.MainCoroutineDispatcher
+import kotlinx.coroutines.launch
 
 class GardenPlantingAdapter(
-    val context: Context
+        val context: Context
 ) : ListAdapter<PlantAndGardenPlantings, GardenPlantingAdapter.ViewHolder>(GardenPlantDiffCallback()) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
         return ViewHolder(
-            DataBindingUtil.inflate(
-                LayoutInflater.from(parent.context),
-                R.layout.list_item_garden_planting, parent, false
-            )
+                DataBindingUtil.inflate(
+                        LayoutInflater.from(parent.context),
+                        R.layout.list_item_garden_planting, parent, false
+                )
         )
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
         getItem(position).let { plantings ->
             with(holder) {
                 itemView.tag = plantings
-                bind(plantings)
+                bind(createOnClickListener(plantings.plant.plantId), createOnLongClickListener(plantings.plant.plantId), plantings)
             }
         }
     }
 
+    private fun createOnClickListener(plantId: String): View.OnClickListener {
+        return View.OnClickListener {
+            val direction =
+                    GardenFragmentDirections.ActionGardenFragmentToPlantDetailFragment(plantId)
+            it.findNavController().navigate(direction)
+        }
+    }
+
+    private fun createOnLongClickListener(plantId: String): View.OnLongClickListener {
+        return View.OnLongClickListener {
+            AlertDialog.Builder(context, R.style.Base_ThemeOverlay_MaterialComponents_Dialog_Alert)
```

Comment:
The whole content of the listener should be handled with `LiveData`, the total clicking will be done in the `fragment`  logical. 
Due to create `Dialog` here, the member `context` should be an `Activity` variant which leaks all, even if `it.context` can be used, the approach to handle click in the adapter is not cool.